### PR TITLE
Added support DingTalk notifier.

### PR DESCRIPTION
**Pull request by mask616**

### CODE REVIEW

**Comment by SteKoe** [comment]
Hi @mask616! Thanks for your contribution. Could you please change the property `webhookUrl` to `webhook-url` as it is in the Slack and MS Teams notifier to have a unique naming? If done, your PR can be merged.

**Comment by mask616** [comment]
> Hi @mask616! Thanks for your contribution. Could you please change the property `webhookUrl` to `webhook-url` as it is in the Slack and MS Teams notifier to have a unique naming? If done, your PR can be merged.

Done. Opened to any feedback.

**Review by SteKoe — APPROVED** [review]
LG2M!

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/notify/DingTalkNotifier.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	 */
+	@Nullable
+	private String secret;
+
+	private Expression message;
+
+	public DingTalkNotifier(InstanceRepository repository, RestTemplate restTemplate) {
+		super(repository);
+		this.restTemplate = restTemplate;
+		this.message = parser.parseExpression(DEFAULT_MESSAGE, ParserContext.TEMPLATE_EXPRESSION);
```
Comment (erikpetzold) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/notify/DingTalkNotifier.java:72]: This is always using DEFAULT_MESSAGE, right? I cannot find any place where the text from property `spring.boot.admin.notify.dingtalk.message` is being used.

**Review by SteKoe — CHANGES_REQUESTED** [review]
Please add some tests as well.

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerNotifierAutoConfiguration.java [hunk 2]
```diff
@@ -309,4 +310,19 @@ public DiscordNotifier discordNotifier(InstanceRepository repository, NotifierPr
 
 	}
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnProperty(prefix = "spring.boot.admin.notify.dingtalk", name = "webhook-url")
```
[the remark this answers is not in the document]
Comment (erikpetzold) [reply spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerNotifierAutoConfiguration.java:314]: ah ok sorry did not see this

**Comment by mask616** [comment]
> Please add some tests as well.

Hi, I've added DingTalk tests.

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/notify/DingTalkNotifier.java (hunk 1 above)
Comment (mask616) [reply spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/notify/DingTalkNotifier.java:72]: This is a reference to other notifiers(eg. TelegramNotifier, SlackNotifier, PagerdutyNotifier, OpsGenieNotifier, LetsChatNotifier...).In oreder to be consistent with others notifiers,  So I did.

Comment (mask616) [reply spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/notify/DingTalkNotifier.java:72]: For example :
In TelegramNotifier
> public TelegramNotifier(InstanceRepository repository, RestTemplate restTemplate) {
		super(repository);
		this.restTemplate = restTemplate;
		this.message = parser.parseExpression(DEFAULT_MESSAGE, ParserContext.TEMPLATE_EXPRESSION);
	}

In SlackNotifier

> public SlackNotifier(InstanceRepository repository, RestTemplate restTemplate) {
		super(repository);
		this.restTemplate = restTemplate;
		this.message = parser.parseExpression(DEFAULT_MESSAGE, ParserContext.TEMPLATE_EXPRESSION);
	}

Comment (erikpetzold) [reply spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/notify/DingTalkNotifier.java:72]: Hi, sorry for answering so late. These other notifiers have a setter for the description where it is parsed:

```
public void setMessage(String message) {
	this.message = parser.parseExpression(message, ParserContext.TEMPLATE_EXPRESSION);
}
```

Comment (erikpetzold) [reply spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/notify/DingTalkNotifier.java:72]: @mask616 if you add this piece we could merge your PR

Comment (mask616) [reply spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/notify/DingTalkNotifier.java:72]: > @mask616 if you add this piece we could merge your PR

@erikpetzold I had add it just now.

**Comment by mask616** [comment]
@SteKoe @erikpetzold Thank you for reviewing my code. I am very happy that my PR can be merged. This is the first time I have participated in open source software. I have learned a lot from this practice. This has also helped me participate in more open source software improvements Gained courage and confidence. Thank you again.

**Comment by erikpetzold** [comment]
Hey @mask616, great to hear 🎉  
Again thanks for your effort and have fun with future contributions

**Pull request merged**